test/RewardsStreamerMP.t.sol

```diff
@@ -1989,3 +1924,121 @@ contract MaliciousUpgradeTest is RewardsStreamerMPTest {
+    }
+
+    function testSetRewards_RevertsBadDuration() public {
+        vm.prank(admin);
+        vm.expectRevert(RewardsStreamerMP.StakingManager__DurationCannotBeZero.selector);
+        streamer.setReward(1000, 0);
+    }
+
+    function testSetRewards_RevertsBadAmount() public {
+        vm.prank(admin);
+        vm.expectRevert(RewardsStreamerMP.StakingManager__AmountCannotBeZero.selector);
+        streamer.setReward(0, 10);
+    }
+
+    function testTotalRewardsSupply() public {
+        _stake(alice, 100e18, 0);
+        assertEq(streamer.totalRewardsSupply(), 0);
+
+        uint256 initialTime = vm.getBlockTimestamp();
+
+        vm.prank(admin);
+        streamer.setReward(1000e18, 10 days);
+        assertEq(streamer.totalRewardsSupply(), 0);
+
+        for (uint256 i = 0; i <= 10; i++) {
+            vm.warp(initialTime + i * 1 days);
+            assertEq(streamer.totalRewardsSupply(), 100e18 * i);
+        }
+
+        // after the end of the reward period, the total rewards supply does not increase
+        vm.warp(initialTime + 11 days);
+        assertEq(streamer.totalRewardsSupply(), 1000e18);
+        assertEq(streamer.totalRewardsAccrued(), 0);
+
+        uint256 secondRewardTime = initialTime + 20 days;
+        vm.warp(secondRewardTime);
+
+        // still the same rewards supply after 20 days
+        assertEq(streamer.totalRewardsSupply(), 1000e18);
+        assertEq(streamer.totalRewardsAccrued(), 0);
+
+        // set other 2000 rewards for other 10 days
+        vm.prank(admin);
+        streamer.setReward(2000e18, 10 days);
+        // accrued is 1000 from the previous reward and still 0 for the new one
+        assertEq(streamer.totalRewardsSupply(), 1000e18, "totalRewardsSupply should be 1000");
+        assertEq(streamer.totalRewardsAccrued(), 1000e18);
+
+        uint256 previousSupply = 1000e18;
+        for (uint256 i = 0; i <= 10; i++) {
+            vm.warp(secondRewardTime + i * 1 days);
+            assertEq(streamer.totalRewardsSupply(), previousSupply + 200e18 * i);
+        }
+    }
+
+    function testRewardsBalanceOf() public {
+        assertEq(streamer.totalRewardsSupply(), 0);
+
+        vm.warp(0);
+
+        uint256 initialTime = vm.getBlockTimestamp();
+
+        _stake(alice, 100e18, 0);
+        assertEq(streamer.rewardsBalanceOf(vaults[alice]), 0);
+
+        vm.prank(admin);
+        streamer.setReward(1000e18, 10 days);
+        assertEq(streamer.rewardsBalanceOf(vaults[alice]), 0);
+
+        vm.warp(initialTime + 1 days);
+
+        // FIXME: this is needed to update the global state and account MP
+        // Later we should update the functions to use "real-time" values.
+        streamer.updateGlobalState();
+        streamer.updateAccountMP(vaults[alice]);
+
+        uint256 tolerance = 300; // 300 wei
+
+        assertEq(streamer.totalRewardsSupply(), 100e18, "Total rewards supply mismatch");
+        assertApproxEqAbs(streamer.rewardsBalanceOf(vaults[alice]), 100e18, tolerance);
```

Comment:
Is this due to rounding issues? If yes, @3esmit brought something similar up in https://github.com/vacp2p/staking-reward-streamer/issues/85

Reply:
yes